proxy/src/http/conn_pool.rs

```diff
@@ -0,0 +1,277 @@
+use pq_proto::StartupMessageParams;
+use std::fmt;
+use std::{collections::HashMap, sync::Arc};
+use tokio::sync::Mutex;
```

Comment:
Is there a specific reason to use an async mutex here? I personally try to avoid them if possible

Reply:
Switched to parking_lot